lib/matplotlib/axes/_axes.py

```diff
@@ -5871,6 +5871,40 @@ def hist(self, x, bins=10, range=None, normed=False, weights=None,
         .. plot:: mpl_examples/statistics/histogram_demo_features.py
 
         """
+        def _normalize_input(inp, ename='input'):
+            """Normalize 1 or 2d input into list of np.ndarray or
+            a single 2D np.ndarray.
```

Comment:
Right, the API is awkward, but I am fine with it as it is private and internal, and it does clean up the code (and fixes a possible mutation side-effect in weights processing). I was more aiming for a clearer docstring for future devs. The if-statements are quite opaque.